src/mapping/PartitionOfUnityMapping.hpp

```diff
@@ -0,0 +1,390 @@
+  PRECICE_DEBUG("Computing cluster-vertex association");
+  for (const auto &vertex : outMesh->vertices()) {
+    auto clusterIDs = clusterIndex.getVerticesInsideBox(vertex, averageClusterRadius);
+    // Consider the case where we didn't find any partition (meshes don't match very well)
+    if (clusterIDs.size() == 0) {
+      PRECICE_WARN("Output vertex {} could not be assigned to a cluster. This means that the meshes probably do not match well geometry-wise.", vertex.getCoords());
+      // TODO: Think about a proper way to handle this case, maybe set all radii to distance(v, closestvertex)?
+      clusterIDs.emplace_back(clusterIndex.getClosestVertex(vertex.getCoords()).index);
+    }
+
+    // Step 4: compute the normalized weights of each output vertex for each partition
+    PRECICE_ASSERT(clusterIDs.size() > 0, "No cluster found for vertex {}", vertex.getCoords());
+
+    // Step 4a: compute the weight in each partition individually and store them in 'weights'
+    std::vector<double> weights(clusterIDs.size());
+    std::transform(clusterIDs.cbegin(), clusterIDs.cend(), weights.begin(), [&](const auto &ids) { return _clusters[ids].computeWeight(vertex); });
+    double weightSum = std::accumulate(weights.begin(), weights.end(), static_cast<double>(0.));
+    // TODO: This covers the edge case of vertices being at the edge of (several) clusters
+    // In case the sum is equal to zero, we assign equal weights for all clusters
+    if (!(weightSum > 0)) {
+      PRECICE_ASSERT(weights.size() > 0);
+      std::for_each(weights.begin(), weights.end(), [&weights](auto &w) { w = 1 / weights.size(); });
+      weightSum = 1;
+    }
+    PRECICE_DEBUG("Weight sum {}", weightSum);
+    PRECICE_DEBUG("Clusters {}", clusterIDs);
+    PRECICE_DEBUG("V coords {}", vertex.getCoords());
+    PRECICE_ASSERT(weightSum > 0);
+
+    // Step 4b: scale the weight using the weight sum and store the normalized weight in all associated clusters
+    for (unsigned int i = 0; i < clusterIDs.size(); ++i) {
+      PRECICE_ASSERT(clusterIDs[i] < _clusters.size());
+      _clusters[clusterIDs[i]].setNormalizedWeight(weights[i] / weightSum, vertex.getID());
+    }
+  }
+
+// Add a VTK export for visualization purposes
+#ifndef NDEBUG
+  exportClusterCentersAsVTU(centerMesh);
+#endif
+
+  // Set the computedMapping flag
+  this->_hasComputedMapping = true;
+  PRECICE_DEBUG("Compute Mapping is Completed.");
+}
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+void PartitionOfUnityMapping<RADIAL_BASIS_FUNCTION_T>::mapConservative(DataID inputDataID, DataID outputDataID)
+{
+  PRECICE_TRACE(inputDataID, outputDataID);
+
+  precice::utils::Event e("map.pou.mapData.From" + input()->getName() + "To" + output()->getName(), precice::syncMode);
+
+  // Execute the actual mapping evaluation in all clusters
+  // 1. Reset all output data values as we accumulate data in all clusters independent
+  output()->data(outputDataID)->values().setZero();
```

Comment:
Could be. For the function here, the setZero is absolutely necessary in order to ensure that it works properly. Do you want to rely on the 'somewhere' reset and transform the above into an assert?
Would require a corresponding reset in the tests, though.

Reply:
Yes, I think it would be better to assert here and check whether things are actually reset. This resetting should be independent of the actual mapping used.